payment/src/main/java/org/killbill/billing/payment/core/PaymentProcessor.java

```diff
@@ -410,11 +410,6 @@ public boolean apply(final PaymentTransactionModelDao input) {
             currentStateName = paymentModelDao.getLastSuccessStateName();
         }
 
-        // Sanity: no paymentMethodId was passed through API and account does not have a default paymentMethodId
-        if (paymentStateContext.getPaymentMethodId() == null) {
-            throw new PaymentApiException(ErrorCode.PAYMENT_NO_DEFAULT_PAYMENT_METHOD, paymentStateContext.getAccount().getId());
```

Comment:
Yes. The Java client should be able to catch it (similar to https://github.com/killbill/killbill/blob/master/profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestAccount.java#L100-L102).